sql: move mutations from planNode to processor model

DrewKimball opened the pull request:
#### sql,rowexec: export/add stats collection helpers

This commit prepares for the mutation operators to be implemented as
local processors by exporting some `rowexec` helpers for execution stats
collection. It also adds some contention/RU event listeners to
`mutationOutputHelper`.

Epic: None
Release note: None

#### sql: move delete from planNode to processor impl

This commit converts deleteNode, deleteSwapNode, and deleteRangeNode from
planNode implementations to LocalProcessor implementations. This change
continues the effort to move away from the old planNode interface for
execution.

Epic: None
Release note: None

#### sql: move update from planNode to processor impl

This commit converts updateNode and updateSwapNode from planNode
implementations to LocalProcessor implementations. This change
continues the effort to move away from the old planNode interface for
execution.

Epic: None
Release note: None

#### sql: move insert/upsert from planNode to processor impl

This commit converts insertNode, upsertNode, and insertFastPathNode from
planNode implementations to LocalProcessor implementations. This change
continues the effort to move away from the old planNode interface for
execution.

Epic: None
Release note: None

#### sql: remove dead mutation code

This commit removes some code for handling mutation operators that is
now unused.

Epic: None
Release note: None

DrewKimball committed:
sql: move delete from planNode to processor impl

This commit converts deleteNode, deleteSwapNode, and deleteRangeNode from
planNode implementations to LocalProcessor implementations. This change
continues the effort to move away from the old planNode interface for
execution.

Epic: None
Release note: None

DrewKimball committed:
sql: move update from planNode to processor impl

This commit converts updateNode and updateSwapNode from planNode
implementations to LocalProcessor implementations. This change
continues the effort to move away from the old planNode interface for
execution.

Epic: None
Release note: None

DrewKimball committed:
sql: move insert/upsert from planNode to processor impl

This commit converts insertNode, upsertNode, and insertFastPathNode from
planNode implementations to LocalProcessor implementations. This change
continues the effort to move away from the old planNode interface for
execution.

Epic: None
Release note: None

DrewKimball commented:
First two commits are #145934.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/150715)
<!-- Reviewable:end -->

DrewKimball committed:
sql,rowexec: export/add stats collection helpers

This commit prepares for the mutation operators to be implemented as
local processors by exporting some `rowexec` helpers for execution stats
collection. It also adds some contention/RU event listeners to
`mutationOutputHelper`.

Epic: None
Release note: None

DrewKimball committed:
sql: remove dead mutation code

This commit removes some code for handling mutation operators that is
now unused.

Epic: None
Release note: None

DrewKimball committed:
sql: qualify type names in DistSQLTypeResolver

Previously, DistSQLTypeResolver.GetTypeDescriptor returned an unqualified
name for the resolved type. The following commits will refactor mutations
to go through the processor infrastructure instead of using the planner
directly, and some error messages will use GetTypeDescriptor during
formatting. With this commit, the error messages will remain schema-qualified
as before.

Epic: None
Release note: None
